### DISCUSSION TITLE
Custom uniform column based layout

### DISCUSSION THREAD

**Discussion by thesobercoder** [ianyh/Amethyst#1483] [Q&A] [answered]
I'm trying to create a custom layout which is very close to the `3Column-Middle` layout but instead of the main pane taking more width than the other two panes, I want the width to be equal. What I'm trying to achieve is something like this - 

![Untitled](https://user-images.githubusercontent.com/1915249/232331993-0be11525-c891-4c35-919e-b9236fc913f2.png)

This image was generated with the Komorebi custom layout generator found [here](https://lgug2z.github.io/komorebi-custom-layout-generator/). Komerebi is a tiling manager for Windows, where I very frequently use the layout shown above.

Any help will be deeply appreciated.

**Comment by cichacz** [comment]
Have you tried using those shortcuts? It allows you to modify size of main pane
<img width="378" alt="image" src="https://user-images.githubusercontent.com/6097288/233029465-bee0cc8c-bc4c-4088-8d7b-36afbe57acc4.png">

**Reply by thesobercoder** [reply]
@cichacz Thanks for pointing that out. On Windows, I'm spoiled by having the `uniform-columns` layout built in that I don't even have to think about it. I was a bit surprised that this was not provided out-of-the-box, although now I downloaded the custom layout and it works as expected. I feel like this should be a built-in layout, no? Or maybe perhaps I'm missing something here around Amethyst philosophy. I feel bad saying `this feature is missing` without understanding why the author chose not to include it by default.

**Reply by ianyh** [reply, marked as answer]
The window manager Amethyst is most inspired by tends to function differently. Regardless, that native layout was a previous contribution from someone so I don't have strong opinions either way, and the apparent disagreement on how it should work is one of the reasons I wanted to move more towards a custom layout system rather than a proliferation of custom options for native ones.

**Reply by thesobercoder** [reply]
@ianyh Thank you for the clarification. I'm guessing at some point, you would want all default layouts to be JavaScript-driven as well.
